example/hydra_plugins/launchers/my_submitit_launcher.py

```diff
@@ -0,0 +1,65 @@
+import logging
+from time import sleep
+from typing import Any
+
+import wandb
+from hydra_plugins.hydra_submitit_launcher.submitit_launcher import BaseSubmititLauncher
+
+logger = logging.getLogger(__name__)
+
+# This has to be located under a hydra_plugins folder or else hydra won't recognize this as a valid plugin.
+# Furthermore, placing an __init__.py under hydra_plugins will cause circular import issues with the above
+# 'from hydra_plugins' import (which is why hydra says not to do this
+# https://hydra.cc/docs/next/advanced/plugins/develop/#internaldocs-banner). However, although Hydra's instantiate
+# function can resolve classes without requiring __init__.py, submitit requires this when pickling. Otherwise
+# submitit won't be able to find the my_submitit_launcher module. A way around this is to make a nested folder,
+# 'launchers', with an __init__.py and place the module there. Then from train.py we register
+# hydra_plugins.launchers by value.
+class MyBaseSubmititLauncher(BaseSubmititLauncher):
```

Comment:
Got any ideas? It's meant to be an example, not a part of the core sweeper code. It's fairly common (and not unpythonic) to use "My" as part of the naming of an example class, function, variable, etc. https://www.python.org/dev/peps/pep-0008/ even has some examples here. Specifically, for this example the naming was inspired from this [GH issue](https://github.com/facebookresearch/hydra/issues/2042#issuecomment-1045333597) I made for Hydra a couple of weeks ago.

Reply:
It's not a core requirement for the submitit integration since that's all up to the user. Basically, I tried making the sweeper as launcher agnostic as possible, despite not testing for the other launchers. The way I did that was by designing the sweeper implementation around the mindset of "What happens if a SystemExit is called by the task function? What if it comes from the agent? What if it comes from Hydra? What if it's a os._exit where it's not an exception and everything just blows up ungracefully (as wandb does in `_on_finish()`)? What happens if signals are being sent to this process? How would I deal with this if wandb already does some signal handling itself?" With this mindset, it's fairly unspecific to SLURM and submitit but it just happens to cover those bases since SLURM uses signalling for pre-emption. If the other cluster-based launchers do the same then great! If not, well, it's up to the user to figure it out and it shouldn't be up to this sweeper. Again, I kept it as launcher agnostic as possible because it's not the sweeper's job to figure out what launcher it's playing with. I know I mention "submitit" in the comments of `_init.py` here and there but really it applies to any cluster-based launcher.

I think testing out the other launchers should be a separate PR.